### DISCUSSION TITLE
FB2 File Support

### DISCUSSION THREAD

**Discussion by kulak** [stumpapp/stump#526] [Question]
How easy would it be to add FB2 file support from internal design perspective?

**Comment by aaronleopold** [comment]
I'm not familiar with that format, but after taking a cursory glance it seems to be a ZIP with a single XML file. Does that sound right?

Implementing it fully would involve:

- Adjust the scanner to accept that format
- Implement the FileProcessor trait for a new FB2 processor
   - Any special considerations for metadata etc
- If epub.js doesn't support the files, we'd either have to build a new reader on the UI or just skip the UI initially

**Reply by kulak** [reply]
I might be interested in adding this format myself, since it turns out I have quite a few FB2 files. It is a question of time.

FB2 tends to be on the smaller side, when compared to epub.  This is due to fewer features compared to epub.

**Reply by aaronleopold** [reply]
If you need any help adding it feel free to reach out! It looks like there is even a specialized crate for it already: https://crates.io/crates/fb2/0.4.1

**Comment by rvalitov** [comment]
Any progress on this? Support of fb2 would be awesome!

**Reply by aaronleopold** [reply]
Hey! No progress has been made. My free time to work on Stump is pretty much solely focused on the big migration I'm trying to finish. When I'm by a computer I'll transfer this discussion to an issue, it'll be easier to track that way

Edit: See https://github.com/stumpapp/stump/issues/714

**Reply by rvalitov** [reply]
Thank you so much! That would be awesome. Fb2 is around 50% of my books collection and FBReader is my favorite reading app. So currently I'm using Calibre Web because of that. Perhaps I should convert my library to EPUB format. Meanwhile I will check if that's possible and works good. But native Fb2 support would be great! 😊
